modules/library/xml/src/main/java/org/geotools/GML.java

```diff
@@ -296,7 +310,9 @@ public void encode(OutputStream out, SimpleFeatureCollection collection) throws
             }
         }
         if (version == Version.WFS1_0) {
-            Encoder e = new Encoder(new org.geotools.wfs.v1_0.WFSConfiguration());
+            org.geotools.wfs.v1_0.WFSConfiguration configuration = new org.geotools.wfs.v1_0.WFSConfiguration();
+            configuration.getProperties().add(GMLConfiguration.OPTIMIZED_ENCODING);
```

Comment:
As your class says in another point, cannot encode a GML2 collection (there is no such a thing)